instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-wildfly-testing/jaxrs-2.0-wildfly-testing.gradle

```diff
@@ -0,0 +1,69 @@
+ext.skipPublish = true
+apply from: "$rootDir/gradle/instrumentation.gradle"
+
+configurations {
+  testServer
+}
+
+dependencies {
+  api "javax:javaee-api:7.0"
+
+  def arquillianVersion = '1.4.0.Final'
+  testImplementation "org.jboss.arquillian.junit:arquillian-junit-container:${arquillianVersion}"
+  testImplementation "org.jboss.arquillian.protocol:arquillian-protocol-servlet:${arquillianVersion}"
+  testImplementation "org.wildfly.arquillian:wildfly-arquillian-container-embedded:2.2.0.Final"
+  testImplementation 'org.jboss.arquillian.spock:arquillian-spock-container:1.0.0.CR1'
+  testImplementation "org.jboss.shrinkwrap.resolver:shrinkwrap-resolver-gradle-depchain:3.1.3"
+  testImplementation "org.glassfish.jersey.core:jersey-client:2.8"
+
+  testInstrumentation project(':instrumentation:servlet:servlet-3.0:javaagent')
+  testInstrumentation project(':instrumentation:jaxrs:jaxrs-2.0:jaxrs-2.0-common:javaagent')
+
+  // wildfly version used to run tests
+  testServer "org.wildfly:wildfly-dist:18.0.0.Final@zip"
+}
+
+// extract wildfly dist, path is used from arquillian.xml
+task setupServer(type: Copy) {
+  from zipTree(configurations.testServer.singleFile)
+  into file('build/server/')
+}
+
+// logback-classic contains /META-INF/services/javax.servlet.ServletContainerInitializer
+// that breaks deploy on embedded wildfly
+// create a copy of logback-classic jar that does not have this file
+task modifyLogbackJar(type: Jar) {
+  doFirst {
+    configurations.configureEach {
+      if (it.name.toLowerCase().endsWith('testruntimeclasspath')) {
+        def logbackJar = it.find { it.name.contains('logback-classic') }
+        from zipTree(logbackJar)
+        exclude(
+          "/META-INF/services/javax.servlet.ServletContainerInitializer"
+        )
+        destinationDirectory = file("$buildDir/tmp")
+        archiveFileName = "logback-classic-modified.jar"
+      }
+    }
+  }
+}
+
+test.dependsOn modifyLogbackJar, setupServer
+
+test {
+  doFirst {
+    // --add-modules is unrecognized on jdk8, ignore it instead of failing
+    jvmArgs "-XX:+IgnoreUnrecognizedVMOptions"
+    // needed for java 11
+    jvmArgs "--add-modules=java.se"
```

Comment:
this seems odd, can you add a bit more explanation to the comment?

Reply:
https://stackoverflow.com/questions/52852192/how-to-run-wildfly-14-with-java-11
wildfly needs java.se module for idk what. I guess in later versions they just add this in their startup scripts, but as we are using embedded wildfly we need to make sure that the environment is suitable for it ourself